code/game/machinery/doors/door.dm

```diff
@@ -76,8 +76,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
+		if(!M.restrained())
+			if(ispet(AM))
+				var/mob/living/simple_animal/A = AM
+				if(A.collar)

```

Comment:
There is a way of putting both checks in one statement, but it's FAR better to have it how you've got it.
Just as an FYI, the way of doing it would be `if(ispet(AM) && AM:collar)`
The : in `AM:collar` tells the compiler you know what you're doing and AM will have a collar, overriding the safeties that would throw the compile error. Which is fine, as if statements work left to right and break out as soon as they fail, right up until the next coder comes along and changes the ispet bit without spotting you used the :, at which time it will start throwing runtimes.

As I said, that was just as an FYI, I can guarantee that if you change it to that the maintainers won't let it through as overriding the compiler safeties is very bad practice